azure/app-service/webapp/variables.tf

```diff
@@ -50,23 +60,32 @@ variable "zip_deploy_file" {
 variable "application_stack" {
   description = "A map detailing the application stack."
   type = object({
+    docker_image_name        = optional(string)
+    docker_registry_url      = optional(string)
+    docker_registry_username = optional(string)
+    docker_registry_password = optional(string)
+    dotnet_version           = optional(string)
+    java_version             = optional(string)
+    node_version             = optional(string)
+    php_version              = optional(string)
+
+    ## Windows Only
     current_stack                = optional(string)
-    docker_container_name        = optional(string)
-    docker_container_registry    = optional(string)
-    docker_container_tag         = optional(string)
     dotnet_core_version          = optional(string)
-    dotnet_version               = optional(string)
+    tomcat_version               = optional(string)
     java_embedded_server_enabled = optional(bool)
-    java_version                 = optional(string)
-    node_version                 = optional(string)
-    php_version                  = optional(string)
     python                       = optional(bool)
-    tomcat_version               = optional(string)
+
+    ## Linux Only
+    go_version          = optional(string)
+    java_server         = optional(string)
+    java_server_version = optional(string)
+    python_version      = optional(string)
+    ruby_version        = optional(string)
   })
   default = {
-    docker_container_name     = "azure-app-service/samples/aspnethelloworld"
-    docker_container_tag      = "latest"
-    docker_container_registry = "mcr.microsoft.com"
+    docker_image_name   = "azure-app-service/samples/aspnethelloworld:latest"
+    docker_registry_url = "https://mcr.microsoft.com"
   }
 }
 
```

Comment:
I feel as though no arguement within the `virtual_application` really accurately could be described as a `key`, so I simply chose not to include one. A key to me is usually like a `name` arguement but since a virtual application has no `name` then I can't see any suitable key. Having an arbritrary key for example is wasteful.

With that said, I was on the fence a little.... perhaps for example `virtual_path` could be the key? I'm open to suggestions but that's my rational

Reply:
I see your point.  virtual_path could be used as a key but it's not really an obvious decision.
Happy to leave as is for now.